core/src/test/scala-3/cats/derived/ApplySuite.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2015 Miles Sabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.derived
+
+import cats.Apply
+import cats.laws.discipline.{ApplyTests, SerializableTests}
+import cats.laws.discipline.SemigroupalTests.Isomorphisms
+
+class ApplySuite extends KittensSuite {
+  import ApplySuite._
+  import TestDefns._
+
+  def testApply(context: String)(using
+      caseClassWOption: Apply[CaseClassWOption],
+      optList: Apply[OptList],
+      // Requires scala 3.1.2
+//      andInt: Apply[AndInt],
+      interleaved: Apply[Interleaved],
+      listBox: Apply[ListBox]
+  ): Unit = {
```

Comment:
We should make this function `inline` to get rid the `using` clauses - see `FunctorSuite` for an example.

Reply:
Oh ok if doesn't work - forget it 😄 
It's just a test anyway.

Reply:
Oh wait - so it did work 👍

Reply:
No, no - it works fine. But I feel there is something wrong with how it works - I guess I'll have to do more digging with newer Scalas. I for some weird reason have to declare new implicits when there was not need to before.